src/gui/guiBox.cpp

```diff
@@ -31,8 +34,82 @@ void GUIBox::draw()
 	if (!IsVisible)
 		return;
 
-	Environment->getVideoDriver()->draw2DRectangle(m_color, AbsoluteRect,
-			&AbsoluteClippingRect);
+	std::vector<s32> inner_bd = {0, 0, 0, 0};
+	std::vector<s32> outer_bd = {0, 0, 0, 0};
+
+	for (unsigned int i = 0; i <= 3; i++) {
+		if (m_borderwidths[i] > 0)
+			outer_bd[i] = m_borderwidths[i];
+		else
+			inner_bd[i] = m_borderwidths[i];
+	}
+
+	v2s32 pos = AbsoluteRect.UpperLeftCorner;
+	v2s32 geom = AbsoluteRect.LowerRightCorner;
```

Comment:
These should rather be called `upperleft` and `lowerright` to not overcomplicate the variable naming.